sphinx/ext/napoleon/__init__.py

```diff
@@ -438,7 +438,11 @@ def _skip_member(app: Sphinx, what: str, name: str, obj: Any,
                         mod_path = cls_path.split('.')
                         cls = functools.reduce(getattr, mod_path, mod)
                     else:
-                        cls = obj.__globals__[cls_path]
+                        if hasattr(obj, '__wrapped__'):
+                            # handle functions decorated by @functools.wrap
+                            cls = obj.__wrapped__.__globals__[cls_path]
```

Comment:
Could you use `sphinx.util.inspect:unwrap()` instead? It is a helper to unwrap decorators.
```
from sphinx.util import inspect

...

cls = inspect.unwrap(obj).__globals__[cls_path]
```